error: use of undeclared identifier 'hint'; did you mean 'link'?

When trying to compile 2.1.1 on FreeBSD I get:

```
/usr/local/libexec/ccache/cc -DMI_MALLOC_OVERRIDE -DMI_SHARED_LIB -DMI_SHARED_LIB_EXPORT -Dmimalloc_EXPORTS -I/wrkdirs/usr/ports/devel/mimalloc/work/mimalloc-2.1.1/include -O2 -pipe  -fstack-protector-strong -fno-strict-aliasing -O2 -pipe  -fstack-protector-strong -fno-strict-aliasing  -DNDEBUG -fPIC -Wall -Wextra -Wno-unknown-pragmas -fvisibility=hidden -Wstrict-prototypes -Wpedantic -Wno-static-in-inline -ftls-model=initial-exec -fno-builtin-malloc -std=gnu11 -MD -MT CMakeFiles/mimalloc.dir/src/prim/prim.c.o -MF CMakeFiles/mimalloc.dir/src/prim/prim.c.o.d -o CMakeFiles/mimalloc.dir/src/prim/prim.c.o -c /wrkdirs/usr/ports/devel/mimalloc/work/mimalloc-2.1.1/src/prim/prim.c
In file included from /wrkdirs/usr/ports/devel/mimalloc/work/mimalloc-2.1.1/src/prim/prim.c:22:
/wrkdirs/usr/ports/devel/mimalloc/work/mimalloc-2.1.1/src/prim/unix/prim.c:172:184: error: use of undeclared identifier 'hint'; did you mean 'link'?
        _mi_warning_message("unable to directly request aligned OS memory (error: %d (0x%x), size: 0x%zx bytes, alignment: 0x%zx, hint address: %p)\n", err, err, size, try_alignment, hint);
                                                                                                                                                                                       ^~~~
                                                                                                                                                                                       link
/usr/include/unistd.h:351:6: note: 'link' declared here
int      link(const char *, const char *);
         ^
1 error generated.
```

The full build log:
[mimalloc-2.1.1.log](https://github.com/microsoft/mimalloc/files/11177067/mimalloc-2.1.1.log)

Environment:
```
FreeBSD 13.1-RELEASE amd64
clang 14.0.5
```

--- a/src/prim/unix/prim.c
+++ b/src/prim/unix/prim.c
@@ -169,7 +169,7 @@ static void* unix_mmap_prim(void* addr, size_t size, size_t try_alignment, int p
       p = mmap(addr, size, protect_flags, flags | MAP_ALIGNED(n), fd, 0);
       if (p==MAP_FAILED || !_mi_is_aligned(p,try_alignment)) { 
         int err = errno;
-        _mi_warning_message("unable to directly request aligned OS memory (error: %d (0x%x), size: 0x%zx bytes, alignment: 0x%zx, hint address: %p)\n", err, err, size, try_alignment, hint);
+        _mi_warning_message("unable to directly request aligned OS memory (error: %d (0x%x), size: 0x%zx bytes, alignment: 0x%zx, hint address: %p)\n", err, err, size, try_alignment, addr);
       }
       if (p!=MAP_FAILED) return p;
       // fall back to regular mmap      

